Microsoft.Toolkit.Uwp.UI.Controls.Input/RichSuggestBox/RichSuggestBox.cs

```diff
@@ -0,0 +1,943 @@
+            }
+
+            TokenSelected.Invoke(this, new RichSuggestTokenSelectedEventArgs
+            {
+                Token = token,
+                Range = selection.GetClone()
+            });
+        }
+
+        private void InvokeTokenHovering(PointerPoint pointer)
+        {
+            var pointerPosition = TransformToVisual(_richEditBox).TransformPoint(pointer.Position);
+            var padding = _richEditBox.Padding;
+            pointerPosition.X += HorizontalOffset - padding.Left;
+            pointerPosition.Y += VerticalOffset - padding.Top;
+            var range = TextDocument.GetRangeFromPoint(pointerPosition, PointOptions.ClientCoordinates);
+            var linkRange = range.GetClone();
+            range.Expand(TextRangeUnit.Character);
+            range.GetRect(PointOptions.None, out var hitTestRect, out _);
+            hitTestRect.X -= hitTestRect.Width;
+            hitTestRect.Width *= 2;
+            if (hitTestRect.Contains(pointerPosition) && linkRange.Expand(TextRangeUnit.Link) > 0 &&
+                TryGetTokenFromRange(linkRange, out var token))
+            {
+                this.TokenHovering.Invoke(this, new RichSuggestTokenHoveringEventArgs
+                {
+                    Token = token,
+                    Range = linkRange,
+                    CurrentPoint = pointer
+                });
+            }
+        }
+
+        private async Task RequestSuggestionsAsync(ITextRange range = null)
+        {
+            string prefix;
+            string query;
+            var queryFound = range == null
+                ? TryExtractQueryFromSelection(out prefix, out query, out range)
+                : TryExtractQueryFromRange(range, out prefix, out query);
+
+            if (queryFound && prefix == _currentPrefix && query == _currentQuery &&
+                range.EndPosition == _currentRange?.EndPosition && _suggestionPopup.IsOpen)
+            {
+                return;
+            }
+
+            CancelIfNotDisposed(this._suggestionRequestedCancellationSource);
+            this._suggestionRequestedCancellationSource = null;
+
+            if (queryFound)
+            {
+                _currentPrefix = prefix;
+                _currentQuery = query;
+                _currentRange = range;
+
+                if (SuggestionsRequested != null)
+                {
+                    using var tokenSource = new CancellationTokenSource();
+                    _suggestionRequestedCancellationSource = tokenSource;
+                    var cancellationToken = tokenSource.Token;
+                    var eventArgs = new SuggestionsRequestedEventArgs { Query = query, Prefix = prefix };
+                    try
+                    {
+                        await SuggestionsRequested.InvokeAsync(this, eventArgs, cancellationToken);
+                    }
+                    catch (OperationCanceledException)
+                    {
+                        eventArgs.Cancel = true;
```

Comment:
If you still need the cancellation source, you should just dispose it in the finally section per docs: https://docs.microsoft.com/dotnet/api/system.threading.cancellationtokensource